### PR TITLE
Initial implementation of virtfs-ext3 (round 2)

### PR DESCRIPTION
**Pull request by movitto**
The previous PR was pretty cluttered, so starting fresh. This should incorporate all the latest feedback

### CODE REVIEW

**Comment by roliveri** [comment]
@movitto I'm sorry, but this doesn't help. At this point in the review, I don't need commits based on code structure and functionality. I want to be able to see the commits that address my specific comments. This new PR doesn't even have the comment trail, so we're better off with the original PR.

When I asked if there was a way to restore the individual commits, I meant the latest commits that were specifically made to address my comments. That would make it much easier to see if/how the comments were addressed, without having to go through all the diffs again.

**Comment by movitto** [comment]
@roliveri I see what your saying about the review continuity but we also have to factor how things are going to look in the final commit log. While this does entail revisiting a few patches, the 'review' aspect should be minimal as functionality is more compontentized per patch (cleanup patches, implementation patches, spec patches, casette patches).

Perhaps we can meet midway, I'll can open incremental PRs for the previous categories (closing this one), that way we can isolate & review each functionality set on its own. Would that work?

**Comment by roliveri** [comment]
@movitto As I said, at this point in the review, it's not about reviewing each functionality set, we've already reviewed most of that. I want to see the commits that address specific comments (or sets of comments). I don't want to have to go through the full set of diffs to see if/how the comments were addressed. As far as I'm concerned, how things look in the commit log is of little or no concern. I think it's OK to squash local commits, before a push, but I should be able to identify the commit(s) associated with each push. I guess what I'm really saying is that you shouldn't rewrite the commit history of the PR.

I don't want us to have to waste too much time on this. Given the original comment trail is lost in this PR, maybe we should just reopen https://github.com/ManageIQ/virtfs-ext3/pull/1#pullrequestreview-20583050 and I'll dig through the diffs. But from now on, please don't squash commits.

**Comment by movitto** [comment]
@roliveri as discussed today I tried to retrieve the original / split PRs but unfortunately it's a bit late. The 'initial' branch was overwritten w/ the squashed PR & the original commits are no where to be found.

I can backport the fixes into the squashed commit, and / or try split that up again (though they will reference different commit hashes). Or we can just proceed w/ this.

File: lib/virtfs/ext3/file.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    end
+
+    def size
+      @de.size
+    end
+
+    def close
+      @de.close
+    end
+  end # class File
```
Comment (roliveri) [inline lib/virtfs/ext3/file.rb]: @movitto The file class also requires a `raw_read` method (I don't understand how read tests can pass without this).
it probably should also implement `close` even if it's a no-op.
In addition, if available, the `atime`, `ctime` and `mtime` methods should be implemented.

**Review by roliveri — CHANGES_REQUESTED** [review]

File: lib/virtfs/ext3/file.rb (hunk 1 above)
Comment (movitto) [reply lib/virtfs/ext3/file.rb]: @roliveri I have a test case reproducing the issue due to the missing 'raw_read' method and am working on a fix. To do so, I'll have to change the ext3 module's File and file_class_methods to behave similarily to nativefs where 'lookup_file' returns an instance of the 'file_obj' which 'sysread' can be called on (or the equivalent mechanism for the ext3 filesystem).

The issue w/ this is that it's in conflict w/ our original goal of having the first pass of this module be a direct mapping of manageiq/vmdb functionality. As it stands the File class instance in created through file_class_methods using the get_file helper which is an exact line-by-line port of the manageiq getFile method (taking into account style fixes and such).

We can roll this enhancement into this PR w/ the others, but I feel it'd be best to go w/ our original strategy, and do the port first, then enhancement the new virtfs-ext3 module w/ new funcitonaility bit by bit. Thoughts?

**Comment by movitto** [comment]
@roliveri after a bit of a run around, I added a few missing classes and was able to implement the 'raw_read' functionality. Specs and cassettes have been updates.

Close is already in place, perhaps the *time methods can be added in a subsequent patch?

**Comment by roliveri** [comment]
@movitto Sure, the *time methods can wait.

So, did you determine how it was working without the `raw_read` method? Was there a hole in the tests?

**Comment by movitto** [comment]
@roliveri yes, I wrote a small little script to verify this functionality. I just added a spec covering this feature

**Pull request merged**